app/src/main/java/com/asfoundation/wallet/ui/barcode/BarcodeCaptureActivity.java

```diff
@@ -212,36 +208,31 @@ private void requestCameraPermission() {
    */
   @Override public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
       @NonNull int[] grantResults) {
-    if (requestCode != RC_HANDLE_CAMERA_PERM) {
-      Log.d(TAG, "Got unexpected permission result: " + requestCode);
+    if (requestCode == RC_HANDLE_CAMERA_PERM) {
+      if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+        Log.d(TAG, "Camera permission granted - initialize the camera source");
+        // we have permission, so create the camerasource
+        boolean autoFocus = true;
+        boolean useFlash = false;
+        createCameraSource(autoFocus, useFlash);
+      } else {
+        Log.e(TAG,
+            "Permission not granted: results len = " + grantResults.length + " Result code = " + (
+                grantResults.length > 0 ? grantResults[0] : "(empty)"));
+        DialogInterface.OnClickListener listener = new DialogInterface.OnClickListener() {
+          public void onClick(DialogInterface dialog, int id) {
+            finish();
+          }
+        };
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Multitracker sample")
```

Comment:
This title doesn't look right

Reply:
Also this logic will show the rational dialog after the user rejects the access to the permission and closes the screen afterwards.
I think this flow isn't the best solution.